### DISCUSSION TITLE
0DTE option orders getting rejected on Wednesday.

### DISCUSSION THREAD

**Discussion by sach27** [ib-api-reloaded/ib_async#90] [Q&A]
I have been trying to develop some strategies on 0DTE options using ib_async, it does work fine for four days a week, but on Wednesday the code breaks.

It gives me the error saying 
```
Error 200, reqId 924: The contract description specified for NDX is ambiguous., contract: Option(symbol='NDX', lastTradeDateOrContractMonth='20241114', strike=20990, right='C', multiplier=100, exchange='SMART', currency='USD')
```
the reason that I have identified is that on Wednesday there are two options trading for 0DTE, one that expires at the end of that trading session, and the other that expires at next day opening, in TWS it is marked as an 'AM' option.

Is there a way to deal whit this ambiguity?

**Comment by mattsta** [comment]
Yeah, the opex contracts act somewhat differently and there's no friday intraday 0DTE NDX available.

If you enable full logging it should print all the alternative possibilities to the log. Just make sure it's all showing with something like:

```python
    import logging                                                                                                                                                                                                                       
    ib_async.util.logToConsole(logging.INFO)   
```

The difference you are probably looking for is the `tradingClass` contract field.

- NDX daily options are trading class `NDXP` except for opex morning AM expiration which are `NDX` (the "regular" ones) (and when there is an AM `NDX` expiration, no other daily contract exists).
- SPX daily options are `SPXW` trading class and opex trading class is `SPX`
- RUT daily options are `RUTW` trading class and opex trading class is `RUT`

Basically you just need to add `tradingClass="NDXP"` to your contract lookup. You can also construct a full regular OCC contract name using `localSymbol="NDXP 241114C21110000"` as the only populated contract field and it should fill out the entire contract too.
